### PR TITLE
Update/full dependency update

### PR DESCRIPTION
**Pull request by KiOui**
### Summary
Updated all dependencies and bumped up version of React-native to latest (0.63.2).

Most significant changes: 
- Rebuild the entire project using a new template for React-native apps as upgrading the old dependencies would not work.
- Device locales are now accessed by using `getLocales` from `react-native-localize` instead of using `DeviceInfo.getDeviceLocale()`.
- Firebase has been updated to the newest version, notifications are now using `@react-native-firebase/messaging` with `messaging().[function]` instead of using `firebase.messaging().[function]`.
- Firebase `onNotification` has been replaced with `onMessage`.
- Snackbar's `title` argument has been replaced with a `text` argument fixing a warning message.
- `DrawerActions` is now imported from `react-navigation-stack` instead of from `react-navigation`.
- `@react-native-firebase/iid` is now a dependency for getting the firebase device id.
- `react-native-sentry` has been replaced with `@sentry/react-native`.
- `react-native-dotenv` is updated and should now be included with `import {ENV variable} from @env` instead of `import {ENV variable} from react-native-dotenv`.
- The `Podfile` is now updated to work with the latest version of React-native, all dependencies are no longer manually linked and will be auto-linked by React
- `AppDelegate.m` has been updated to work with the latest version of Xcode.
- `Animated.event` now must have a second argument.
- The Android build CI is fixed and is now using Jetifier.

### How to test
All screens must be tested as many things have changed, both iOS and Android builds must be tested.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,8 +1,7 @@
 {
     "parser": "babel-eslint",
     "env": {
-        "browser": true,
-        "es6": true
```
Comment (se-bastiaan) [inline .eslintrc.json]: Don't you want ES6 here?

File: android/app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -41,26 +41,12 @@
...
-                <action android:name="com.google.firebase.MESSAGING_EVENT" />
-            </intent-filter>
-        </service>
         <service android:name="io.invertase.firebase.messaging.RNFirebaseInstanceIdService">
             <intent-filter>
                 <action android:name="com.google.firebase.INSTANCE_ID_EVENT"/>
             </intent-filter>
         </service>
 
-        <meta-data
```
Comment (se-bastiaan) [inline android/app/src/main/AndroidManifest.xml:55]: Do we not need this anymore?

File: android/build.gradle [hunk 3]
```diff
@@ -1,38 +1,38 @@
...
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:3.3.2'
-        classpath 'com.google.gms:google-services:4.2.0'
-
+        classpath("com.android.tools.build:gradle:3.5.4")
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
+        classpath 'com.google.gms:google-services:+'
```
Comment (se-bastiaan) [inline android/build.gradle:18]: You probably want to put a version here

File: android/gradlew [hunk 4]
```diff
@@ -28,7 +44,7 @@ APP_NAME="Gradle"
 APP_BASE_NAME=`basename "$0"`
 
 # Add default JVM options here. You can also use JAVA_OPTS and GRADLE_OPTS to pass JVM options to this script.
-DEFAULT_JVM_OPTS=""
+DEFAULT_JVM_OPTS='"-Xmx64m" "-Xms64m"'
```
Comment (se-bastiaan) [inline android/gradlew:47]: Is this a default value? Thats not much RAM

File: app/utils/url.js [hunk 5]
```diff
@@ -1,16 +1,18 @@
-import DeviceInfo from 'react-native-device-info';
+/* eslint-disable max-classes-per-file */
+import { getLocales } from 'react-native-localize';
+import { Platform } from 'react-native';
 
 let server = 'https://thalia.nu';
 /* istanbul ignore next line */
 if (__DEV__) { // eslint-disable-line no-undef
-  server = 'http://localhost:8000';
+  server = Platform.OS === 'android' ? 'http://10.0.2.2:8000' : 'http://localhost:8000';
```
Comment (se-bastiaan) [inline app/utils/url.js]: Why that IP address?

File: package.json [hunk 6]
```diff
@@ -38,67 +38,77 @@
...
-    "redux-saga-test-plan": "3.7.0"
+    "@babel/core": "^7.11.6",
+    "@babel/plugin-proposal-class-properties": "^7.10.4",
+    "@babel/plugin-proposal-optional-chaining": "^7.11.0",
+    "@babel/plugin-transform-flow-strip-types": "^7.10.4",
+    "@babel/plugin-transform-runtime": "^7.11.5",
+    "@babel/preset-env": "^7.11.5",
+    "@babel/preset-react": "^7.10.4",
+    "@babel/runtime": "^7.11.2",
+    "@react-native-community/eslint-config": "^2.0.0",
```
Comment (se-bastiaan) [inline package.json]: Do we still need the airbnb config with this one already installed?

**Review by se-bastiaan — COMMENTED** [review]
.

File: app/utils/url.js (hunk 5 above)
Comment (KiOui) [reply app/utils/url.js]: On Android localhost will now point to the device itself instead of to the computer you are running it on. The computer will always have 10.0.0.2 as IP address so we have to replace localhost by 10.0.0.2 in development but only for the android side of things.

File: android/app/build.gradle [hunk 7]
```diff
@@ -37,10 +133,9 @@ android {
         applicationId "com.thaliapp"
         minSdkVersion rootProject.ext.minSdkVersion
         targetSdkVersion rootProject.ext.targetSdkVersion
-        versionCode 437
-        versionName "2.8.2"
+        versionCode 1
+        versionName "1.0"
```
Comment (KiOui) [inline android/app/build.gradle]: These should be reverted to their original version.

File: android/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (KiOui) [reply android/app/src/main/AndroidManifest.xml:55]: This causes the app build to crash, I did not test whether notifications still work and don't know how to. Let's test this next Monday.

File: android/gradlew (hunk 4 above)
Comment (KiOui) [reply android/gradlew:47]: Hmm I did not edit this file so I guess this is default, I actually had to increase the ram for Java already so I might remove this again.

File: package.json (hunk 6 above)
Comment (KiOui) [reply package.json]: I'm not sure, let's also discuss this Monday.

File: android/gradlew (hunk 4 above)
Comment (se-bastiaan) [reply android/gradlew:47]: Well it is limiting grade memory to 64MiB and removing it should not break anything

File: android/app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (se-bastiaan) [reply android/app/src/main/AndroidManifest.xml:55]: It is perfectly possible that this is now automatically included in the manifest  due to the new dependencies

File: app/utils/url.js (hunk 5 above)
Comment (se-bastiaan) [reply app/utils/url.js]: Huh? Always 10.0.0.2? That is new to me. The idea of localhost is that the adb will route the 8000 port via a reverse proxy to the device. I suspect that the gradle changes removed that line on the before run though.

File: .gitignore [hunk 8]
```diff
@@ -46,6 +46,8 @@ package-lock.json
 buck-out/
 \.buckd/
 *.keystore
+# But ignore debug.keystore as we use it in the CI
+!android/app/debug.keystore
```
Comment (se-bastiaan) [inline .gitignore:50]: Technically we should not need to provide a debug key store in our code. The SDK creates its own.

File: android/app/build.gradle [hunk 9]
```diff
@@ -49,52 +144,73 @@ android {
...
+            signingConfig signingConfigs.debug
+        }
         release {
+            // Caution! In production, you need to generate your own keystore file.
+            // see https://reactnative.dev/docs/signed-apk-android.
+            signingConfig signingConfigs.debug
             minifyEnabled enableProguardInReleaseBuilds
             proguardFiles getDefaultProguardFile("proguard-android.txt"), "proguard-rules.pro"
         }
-        debug {
```
Comment (se-bastiaan) [inline android/app/build.gradle]: This was fine

File: android/app/build.gradle [hunk 10]
```diff
@@ -49,52 +144,73 @@ android {
...
     implementation fileTree(dir: "libs", include: ["*.jar"])
-    implementation "com.android.support:appcompat-v7:${rootProject.ext.supportLibVersion}"
-    implementation "com.android.support:design:${rootProject.ext.supportLibVersion}"
+    //noinspection GradleDynamicVersion
     implementation "com.facebook.react:react-native:+"  // From node_modules
-    implementation "com.google.firebase:firebase-core:16.0.9"
-    implementation "com.google.firebase:firebase-messaging:18.0.0"
+
+    /* Needed for react-native status bar */
+    implementation project(':@react-native-community_status-bar')
```
Comment (se-bastiaan) [inline android/app/build.gradle:106]: Why is this needed for this lib but not the rest?

File: android/app/build.gradle [hunk 11]
```diff
@@ -104,12 +220,4 @@ task copyDownloadableDepsToLibs(type: Copy) {
     into 'libs'
 }
 
-gradle.projectsEvaluated {
```
Comment (se-bastiaan) [inline android/app/build.gradle]: This should not be removed since it disables the reverse proxy from your localhost to the phone for the running concrexit install. Also see my comment on the url util file.

File: android/app/proguard-rules.pro [hunk 12]
```diff
@@ -8,63 +8,3 @@
 #   http://developer.android.com/guide/developing/tools/proguard.html
 
 # Add any project specific keep options here:
-
```
Comment (se-bastiaan) [inline android/app/proguard-rules.pro:11]: I think we still need this if we want to minify our build

File: android/gradlew (hunk 4 above)
Comment (KiOui) [reply android/gradlew:47]: This is the default file from the react-native repo so let's just leave it like this.

**Review by se-bastiaan — DISMISSED** [review]

**Review by se-bastiaan — APPROVED** [review]
